src/core/folder.cpp

```diff
@@ -129,7 +129,8 @@ bool Folder::isLoaded() const {
 }
 
 std::shared_ptr<const FileInfo> Folder::fileByName(const char* name) const {
-    auto it = files_.find(name);
+    std::string nameStr(name);
+    auto it = files_.find(nameStr);
```

Comment:
There is no need to create the temp `std::string` here. The `std::unordered_map::find(const Key &)` can handle it itself.